graylog2-server/src/main/java/org/graylog2/configuration/MongoDbConfiguration.java

```diff
@@ -173,11 +173,11 @@ public MongoClientURI getMongoClientURI() {
 
     @ValidatorMethod
     public void validate() throws ValidationException {
-        if((isNullOrEmpty(getHost()) || isNullOrEmpty(getDatabase())) && isNullOrEmpty(getUri())) {
+        if ((isNullOrEmpty(getHost()) || isNullOrEmpty(getDatabase()) || getReplicaSet() == null) && isNullOrEmpty(getUri())) {
```

Comment:
I guess this should be `if ((isNullOrEmpty(getHost()) && getReplicaSet() == null) || isNullOrEmpty(getDatabase()) && isNullOrEmpty(getUri())) {` instead. Bonus points for checking that the replica set is not empty.